source/renderer/app/components/wallet/send-form/WalletSendAssetsConfirmationDialog.js

```diff
@@ -260,43 +157,60 @@ export default class WalletSendAssetsConfirmationDialog extends Component<
     (this.props.isHardwareWallet || this.form.$('passphrase').isValid) &&
     submitOnEnter(this.submit, event);
 
-  renderConfirmationElement = (isHardwareWallet: boolean) => {
+  renderConfirmationElement = (
+    isHardwareWallet: boolean
+  ): React$Element<*> | null => {
     const passphraseField = this.form.$('passphrase');
     const { areTermsAccepted } = this.state;
     const {
       hwDeviceStatus,
       isFlight,
+      totalAmount,
       onExternalLinkClick,
-      walletName,
+      wallet,
       isTrezor,
     } = this.props;
 
+    let returnJSX = null;
     if (!isFlight || (isFlight && areTermsAccepted)) {
-      if (isHardwareWallet) {
-        return (
+      const walletWarning = (
+        <div className={styles.flightCandidateWarning}>
+          <FormattedHTMLMessage {...messages.emptyingWarning} tagName="p" />
+        </div>
+      );
+      const { name, isLegacy } = wallet;
+      returnJSX = isHardwareWallet ? (
+        <>
           <div className={styles.hardwareWalletStatusWrapper}>
             <HardwareWalletStatus
               hwDeviceStatus={hwDeviceStatus}
-              walletName={walletName}
+              walletName={name}
               isTrezor={isTrezor}
               onExternalLinkClick={onExternalLinkClick}
             />
           </div>
-        );
-      }
-      return (
-        <Input
-          type="password"
-          className={styles.passphrase}
-          {...passphraseField.bind()}
-          error={passphraseField.error}
-          skin={InputSkin}
-          onKeyPress={this.handleSubmitOnEnter}
-          autoFocus
-        />
+          {!isLegacy &&
+            shouldShowEmptyWalletWarning(totalAmount, wallet) &&
+            walletWarning}
+        </>
+      ) : (
+        <>
+          <Input
+            type="password"
+            className={styles.passphrase}
+            {...passphraseField.bind()}
+            error={passphraseField.error}
+            skin={InputSkin}
+            onKeyPress={this.handleSubmitOnEnter}
+            autoFocus
+          />
+          {!isLegacy &&
```

Comment:
@danielmain can we please avoid this duplication? Perhaps by extracting the walletWarning injection out of the "isHardwareWallet" if clause?